src/fpm_targets.f90

```diff
@@ -307,14 +307,30 @@ subroutine build_target_list(targets,model)
                                     output_name = join_path(exe_dir, &
                                     sources(i)%exe_name//xsuffix))
 
+
+                    associate(target=>targets(size(targets))%ptr)
+
+                    ! If the main program is on a C/C++ source, the Intel Fortran compiler requires option
+                    ! -nofor-main to avoid "duplicate main" errors.
+                    ! https://stackoverflow.com/questions/36221612/p3dfft-compilation-ifort-compiler-error-multiple-definiton-of-main
+                    if (model%compiler%is_intel() .and. any(exe_type==[FPM_TARGET_C_OBJECT,FPM_TARGET_CPP_OBJECT])) then
+                       if (get_os_type()==OS_WINDOWS) then
+                           target%compile_flags = '/nofor-main'
+                       else
+                           target%compile_flags = '-nofor-main'
+                       end if
+                    end if
+
```

Comment:
I think this snippet was supposed to be a part of https://github.com/fortran-lang/fpm/pull/896

Reply:
Correct, was a placeholder until the actual fix. Replaced with the merged code now